src/quota_aggregator_impl.cc

```diff
@@ -137,6 +142,18 @@ ::google::protobuf::util::Status QuotaAggregatorImpl::Quota(
   // Aggregate tokens if the cached response is positive
   if (lookup.value()->is_positive_response()) {
     lookup.value()->Aggregate(request);
+  } else {
+    if (lookup.value()->in_flight() == false) {
+      int64_t elapsed = (SimpleCycleTimer::Now() - lookup.value()->last_refresh_time())
+          / 1000;
+      if (elapsed > options_.refresh_interval_ms) {
+        // update in_flight to avoid duplicated request
+        lookup.value()->set_in_flight(true);
+
+        // trigger a new quota refresh request
+        return Status(Code::NOT_FOUND, "");
```

Comment:
We can use AddRemovedItem() to send refresh.  Not to using returning NOT_FOUND.

Just return the cached response.